src/sss_client/nss_mc_common.c

```diff
@@ -185,6 +185,9 @@ static errno_t sss_nss_mc_init_ctx(const char *name,
     ret = 0;
 
 done:
+    if (ctx->fd) {
+        close(ctx->fd);
```

Comment:
At the very least you must set ctx->fd to -1 after a close, or you'll cause trouble later (closing unrelated files if the context is freed).

Reply:
If you are going to always close the file, which is a fine choice, then I would expect this patch to actually remove the fd element from the struct and use a local variable instead so that no manipulation of the fd is expected in sss_nss_mc_destroy_ctx() at all.